app_dart/lib/src/model/firestore/task.dart

```diff
@@ -211,6 +212,22 @@ class Task extends Document {
     _setStatusFromLuciStatus(build);
   }
 
+  void updateFromBuildV2(bbv2.Build build) {
```

Comment:
Let's add a test to cover the logic here.

Reply:
There are tests in the firestore_test.dart file but I can move them here.

Reply:
Moved the tests.